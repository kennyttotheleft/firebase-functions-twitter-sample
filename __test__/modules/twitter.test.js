'use strict'
jest.mock('../../modules/twitter')
const settings = require('../../config/settings.json')
const functions = require('firebase-functions')
const Twitter = require('twitter')
const TwitterModule = require('../../modules/twitter')

const config = functions.config()
const twitter = new TwitterModule({
    twitterClient: new Twitter({
        consumer_key: config.credential.twitter.consumer_key,
        consumer_secret: config.credential.twitter.consumer_secret,
        access_token_key: config.credential.twitter.access_token_key,
        access_token_secret: config.credential.twitter.access_token_secret
    }),
    cache: require('memory-cache')
})

describe('twiiter.js', () => {

    test('credentials are set for retrieving tweets', () => {
        expect(config.credential.twitter.consumer_key).not.toBeNull()
        expect(config.credential.twitter.consumer_secret).not.toBeNull()
        expect(config.credential.twitter.access_token_key).not.toBeNull()
        expect(config.credential.twitter.access_token_secret).not.toBeNull()
    })

    test('settings are set for retrieving tweets', () => {
        expect(settings.FAVOLITE_TWEET_LIST_OPTIONS.screen_name).not.toBeNull()
    })

    test('get favolite tweet list without options', done => {
        twitter.getFavList(false, (tweets, error) => {
            expect(error).toBeFalsy()
            expect(error).toMatchSnapshot()
            expect(tweets).toMatchSnapshot()
            expect(tweets.length).toBeGreaterThan(0)
            done()
        })
    })

    test('get favolite tweet list with options', done => {
        const options = settings.FAVOLITE_TWEET_LIST_OPTIONS
        twitter.getFavList(options, (tweets, error) => {
            expect(error).toBeFalsy()
            expect(error).toMatchSnapshot()
            expect(tweets).toMatchSnapshot()
            expect(tweets.length).toBeGreaterThan(0)
            done()
        })
    })

})