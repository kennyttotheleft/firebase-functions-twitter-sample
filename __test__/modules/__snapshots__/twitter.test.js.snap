// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`twiiter.js お気に入り Tweet リストが取得できること（取得設定オプションあり） 1`] = `false`;

exports[`twiiter.js お気に入り Tweet リストが取得できること（取得設定オプションあり） 2`] = `
Array [
  Object {
    "contributors": null,
    "coordinates": null,
    "created_at": "Tue Sep 04 15:55:52 +0000 2012",
    "entities": Object {
      "hashtags": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    },
    "favorited": true,
    "geo": null,
    "id": 243014525132091400,
    "id_str": "243014525132091393",
    "in_reply_to_screen_name": null,
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "place": Object {
      "attributes": Object {},
      "bounding_box": Object {
        "coordinates": Array [
          Array [
            Array [
              -122.51368188,
              37.70813196,
            ],
            Array [
              -122.35845384,
              37.70813196,
            ],
            Array [
              -122.35845384,
              37.83245301,
            ],
            Array [
              -122.51368188,
              37.83245301,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "country": "United States",
      "country_code": "US",
      "full_name": "San Francisco, CA",
      "id": "5a110d312052166f",
      "name": "San Francisco",
      "place_type": "city",
      "url": "http://api.twitter.com/1/geo/id/5a110d312052166f.json",
    },
    "retweet_count": 0,
    "retweeted": false,
    "source": "web",
    "text": "Note to self:  don't die during off-peak hours on a holiday weekend.",
    "truncated": false,
    "user": Object {
      "contributors_enabled": true,
      "created_at": "Sat May 09 17:58:22 +0000 2009",
      "default_profile": false,
      "default_profile_image": false,
      "description": "I taught your phone that thing you like.  The Mobile Partner Engineer @Twitter. ",
      "entities": Object {
        "description": Object {
          "urls": Array [],
        },
      },
      "favourites_count": 594,
      "follow_request_sent": false,
      "followers_count": 10659,
      "following": true,
      "friends_count": 1186,
      "geo_enabled": true,
      "id": 38895958,
      "id_str": "38895958",
      "is_translator": false,
      "lang": "en",
      "listed_count": 191,
      "location": "San Francisco",
      "name": "Sean Cook",
      "notifications": false,
      "profile_background_color": "1A1B1F",
      "profile_background_image_url": "http://a0.twimg.com/profile_background_images/495742332/purty_wood.png",
      "profile_background_image_url_https": "https://si0.twimg.com/profile_background_images/495742332/purty_wood.png",
      "profile_background_tile": true,
      "profile_image_url": "http://a0.twimg.com/profile_images/1751506047/dead_sexy_normal.JPG",
      "profile_image_url_https": "https://si0.twimg.com/profile_images/1751506047/dead_sexy_normal.JPG",
      "profile_link_color": "2FC2EF",
      "profile_sidebar_border_color": "181A1E",
      "profile_sidebar_fill_color": "252429",
      "profile_text_color": "666666",
      "profile_use_background_image": true,
      "protected": false,
      "screen_name": "theSeanCook",
      "show_all_inline_media": true,
      "statuses_count": 2629,
      "time_zone": "Pacific Time (US & Canada)",
      "url": null,
      "utc_offset": -28800,
      "verified": false,
    },
  },
  Object {
    "contributors": null,
    "coordinates": null,
    "created_at": "Tue Sep 04 00:17:11 +0000 2012",
    "entities": Object {
      "hashtags": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    },
    "favorited": true,
    "geo": null,
    "id": 242778296117514240,
    "id_str": "242778296117514240",
    "in_reply_to_screen_name": null,
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "place": null,
    "retweet_count": 1,
    "retweeted": false,
    "source": "Twitter for Android",
    "text": "TWIT NPC. TWIT DUNGEONMASTER.",
    "truncated": false,
    "user": Object {
      "contributors_enabled": false,
      "created_at": "Wed May 07 19:27:16 +0000 2008",
      "default_profile": false,
      "default_profile_image": false,
      "description": "Everything I tweet is or shall be.",
      "entities": Object {
        "description": Object {
          "urls": Array [],
        },
      },
      "favourites_count": 46407,
      "follow_request_sent": false,
      "followers_count": 4198,
      "following": true,
      "friends_count": 200,
      "geo_enabled": true,
      "id": 14690653,
      "id_str": "14690653",
      "is_translator": false,
      "lang": "en",
      "listed_count": 194,
      "location": "",
      "name": "REGIS",
      "notifications": false,
      "profile_background_color": "131516",
      "profile_background_image_url": "http://a0.twimg.com/profile_background_images/530267284/11111.JPG",
      "profile_background_image_url_https": "https://si0.twimg.com/profile_background_images/530267284/11111.JPG",
      "profile_background_tile": false,
      "profile_image_url": "http://a0.twimg.com/profile_images/1812284389/allseeingeye_normal.jpg",
      "profile_image_url_https": "https://si0.twimg.com/profile_images/1812284389/allseeingeye_normal.jpg",
      "profile_link_color": "009999",
      "profile_sidebar_border_color": "C0DEED",
      "profile_sidebar_fill_color": "DDEEF6",
      "profile_text_color": "333333",
      "profile_use_background_image": true,
      "protected": false,
      "screen_name": "regisl",
      "show_all_inline_media": false,
      "statuses_count": 43345,
      "time_zone": "Mountain Time (US & Canada)",
      "url": null,
      "utc_offset": -25200,
      "verified": false,
    },
  },
]
`;

exports[`twiiter.js お気に入り Tweet リストが取得できること（取得設定オプションなし） 1`] = `false`;

exports[`twiiter.js お気に入り Tweet リストが取得できること（取得設定オプションなし） 2`] = `
Array [
  Object {
    "contributors": null,
    "coordinates": null,
    "created_at": "Tue Sep 04 15:55:52 +0000 2012",
    "entities": Object {
      "hashtags": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    },
    "favorited": true,
    "geo": null,
    "id": 243014525132091400,
    "id_str": "243014525132091393",
    "in_reply_to_screen_name": null,
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "place": Object {
      "attributes": Object {},
      "bounding_box": Object {
        "coordinates": Array [
          Array [
            Array [
              -122.51368188,
              37.70813196,
            ],
            Array [
              -122.35845384,
              37.70813196,
            ],
            Array [
              -122.35845384,
              37.83245301,
            ],
            Array [
              -122.51368188,
              37.83245301,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "country": "United States",
      "country_code": "US",
      "full_name": "San Francisco, CA",
      "id": "5a110d312052166f",
      "name": "San Francisco",
      "place_type": "city",
      "url": "http://api.twitter.com/1/geo/id/5a110d312052166f.json",
    },
    "retweet_count": 0,
    "retweeted": false,
    "source": "web",
    "text": "Note to self:  don't die during off-peak hours on a holiday weekend.",
    "truncated": false,
    "user": Object {
      "contributors_enabled": true,
      "created_at": "Sat May 09 17:58:22 +0000 2009",
      "default_profile": false,
      "default_profile_image": false,
      "description": "I taught your phone that thing you like.  The Mobile Partner Engineer @Twitter. ",
      "entities": Object {
        "description": Object {
          "urls": Array [],
        },
      },
      "favourites_count": 594,
      "follow_request_sent": false,
      "followers_count": 10659,
      "following": true,
      "friends_count": 1186,
      "geo_enabled": true,
      "id": 38895958,
      "id_str": "38895958",
      "is_translator": false,
      "lang": "en",
      "listed_count": 191,
      "location": "San Francisco",
      "name": "Sean Cook",
      "notifications": false,
      "profile_background_color": "1A1B1F",
      "profile_background_image_url": "http://a0.twimg.com/profile_background_images/495742332/purty_wood.png",
      "profile_background_image_url_https": "https://si0.twimg.com/profile_background_images/495742332/purty_wood.png",
      "profile_background_tile": true,
      "profile_image_url": "http://a0.twimg.com/profile_images/1751506047/dead_sexy_normal.JPG",
      "profile_image_url_https": "https://si0.twimg.com/profile_images/1751506047/dead_sexy_normal.JPG",
      "profile_link_color": "2FC2EF",
      "profile_sidebar_border_color": "181A1E",
      "profile_sidebar_fill_color": "252429",
      "profile_text_color": "666666",
      "profile_use_background_image": true,
      "protected": false,
      "screen_name": "theSeanCook",
      "show_all_inline_media": true,
      "statuses_count": 2629,
      "time_zone": "Pacific Time (US & Canada)",
      "url": null,
      "utc_offset": -28800,
      "verified": false,
    },
  },
  Object {
    "contributors": null,
    "coordinates": null,
    "created_at": "Tue Sep 04 00:17:11 +0000 2012",
    "entities": Object {
      "hashtags": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    },
    "favorited": true,
    "geo": null,
    "id": 242778296117514240,
    "id_str": "242778296117514240",
    "in_reply_to_screen_name": null,
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "place": null,
    "retweet_count": 1,
    "retweeted": false,
    "source": "Twitter for Android",
    "text": "TWIT NPC. TWIT DUNGEONMASTER.",
    "truncated": false,
    "user": Object {
      "contributors_enabled": false,
      "created_at": "Wed May 07 19:27:16 +0000 2008",
      "default_profile": false,
      "default_profile_image": false,
      "description": "Everything I tweet is or shall be.",
      "entities": Object {
        "description": Object {
          "urls": Array [],
        },
      },
      "favourites_count": 46407,
      "follow_request_sent": false,
      "followers_count": 4198,
      "following": true,
      "friends_count": 200,
      "geo_enabled": true,
      "id": 14690653,
      "id_str": "14690653",
      "is_translator": false,
      "lang": "en",
      "listed_count": 194,
      "location": "",
      "name": "REGIS",
      "notifications": false,
      "profile_background_color": "131516",
      "profile_background_image_url": "http://a0.twimg.com/profile_background_images/530267284/11111.JPG",
      "profile_background_image_url_https": "https://si0.twimg.com/profile_background_images/530267284/11111.JPG",
      "profile_background_tile": false,
      "profile_image_url": "http://a0.twimg.com/profile_images/1812284389/allseeingeye_normal.jpg",
      "profile_image_url_https": "https://si0.twimg.com/profile_images/1812284389/allseeingeye_normal.jpg",
      "profile_link_color": "009999",
      "profile_sidebar_border_color": "C0DEED",
      "profile_sidebar_fill_color": "DDEEF6",
      "profile_text_color": "333333",
      "profile_use_background_image": true,
      "protected": false,
      "screen_name": "regisl",
      "show_all_inline_media": false,
      "statuses_count": 43345,
      "time_zone": "Mountain Time (US & Canada)",
      "url": null,
      "utc_offset": -25200,
      "verified": false,
    },
  },
]
`;
